lib/ws/ws.js

```diff
@@ -9,13 +9,86 @@ var factory = require('./factory');
 var authMethod = 'auth';
 
 /**
+ * race takes an array of promises and is resolved with the first promise
+ * that resolves, or rejects. After the first resolve or rejection, all
+ * future resolutions and rejections will be discarded.
+ * @param {Promise[]} promises
+ * @return {Promise}
+ */
+function race(promises) {
```

Comment:
Bluebird is a subdependency of several modules, including beam-client-node, but is excluded from the webpack bundle.